src/overlays/actors/ovl_En_Gg/z_en_gg.c

```diff
@@ -74,62 +95,719 @@ static DamageTable D_80B36C38 = {
+                    } while (0);
+                    this->unk_2E6 = 14;
+                    func_80B364D4(&this->unk_344, globalCtx);
+                    break;
+
+                case 8:
+                    Actor_MarkForDeath(&this->actor);
+                    return;
+
+                default:
+                    this->unk_2DA = 0;
+                    break;
+            }
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_2DA);
+        }
+
+        if (this->unk_2DA == 14) {
+            func_80B358D8(this, globalCtx);
+        }
+
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+        this->actor.shape.yOffset = 0.0f;
+    } else {
+        this->unk_2DB = 99;
+    }
+}
+
+u16 func_80B357F0(EnGg* this) {
+    switch (this->actor.textId) {
+        case 0xCE5:
+            return 0xCE6;
+        case 0xCE6:
+            return 0xCE8;
+        case 0xCE8:
+            return 0xCE9;
+        case 0xCE9:
+            return 0xCEC;
+        case 0xCEC:
+            return 0xCED;
+    }
+    return 0;
+}
+
+void func_80B3584C(EnGg* this) {
+    this->unk_30A += 0x72C;
+    this->actor.shape.yOffset = Math_SinS(this->unk_30A) * 100.0f;
+    this->actor.shape.shadowScale = 30.0f - (Math_SinS(this->unk_30A) * 5.0f);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+void func_80B358D8(EnGg* this, GlobalContext* globalCtx) {
+    s16 sp1E = this->skelAnime.curFrame;
+    s16 frame = Animation_GetLastFrame(sAnimations[this->unk_2DA].animation);
+
+    if ((this->unk_2E6 == 14) && (sp1E == frame)) {
+        this->unk_2E6 = 15;
+        Actor_ChangeAnimation(&this->skelAnime, sAnimations, 15);
+    }
+}
+
+void func_80B35968(EnGg* this, GlobalContext* globalCtx) {
+    if (this->unk_344.unk_34 != NULL) {
+        this->unk_344.unk_34(&this->unk_344, globalCtx);
+    } else {
+        if ((this->unk_2DA == 11) || (this->unk_2DA == 14)) {
+            this->unk_344.unk_30 = this->unk_2DA;
+        }
+        func_80B363E8(&this->unk_344, globalCtx, &this->unk_320, &this->unk_32C, &this->unk_338);
+    }
+}
+
+void func_80B359DC(EnGg* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (this->actor.xzDistToPlayer < 200.0f) {
+        if (this->unk_306 == 0) {
+            if (player->stateFlags2 & 0x8000000) {
+                this->unk_306 = 1;
+                play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            }
+        } else if (!(player->stateFlags2 & 0x8000000)) {
+            this->unk_306 = 0;
+        }
+
+        if ((player->transformation == PLAYER_FORM_HUMAN) && (globalCtx->msgCtx.unk1202A == 3) &&
+            (globalCtx->msgCtx.unk1202E == 7)) {
+            if (!(gSaveContext.weekEventReg[19] & 0x80)) {
+                gSaveContext.weekEventReg[19] |= 0x80;
+            }
+            this->unk_307 = 1;
```

Comment:
```suggestion
            this->unk_307 = true;
```
This is boolean, you can apply this to the rest of the function